File: Cypress/cypress/e2e/engineerDegree/testPierwszySklepPW.js
describe('PW', () => {
  beforeEach(() => {
   cy.visit('https://www.sklep.pw.edu.pl/',{
    })
   })

  it('add item to the basket and then clear it', () => {
      cy.get('.search-field').eq(0).type('koszulka').type('{enter}')
      cy.get('.add_to_cart_button').eq(1).click()
      cy.get('[id^=pa_rodzaj]').select(1)
      cy.get('[id^=pa_rozmiar]').select(1)

      cy.contains('Add to cart').click()
      cy.contains('has been added to your cart')
      cy.get('.woocommerce-message > .button').click()
      cy. contains('Cart')
      cy.get('.remove').click()
      cy.contains('Your cart is currently empty.')
  })
})

File: Selenium/testPierwszySklepPW.java
package politechnikaWarszawska;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

public class testPierwszySklepPW {
    WebDriver driver;
    private WebElement wyszukiwarka;

    @Test
    public void testPierwszySklepPW() {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--lang=pl");
        options.addArguments("--start-maximized");
        WebDriverManager.chromedriver().setup();
        driver = new ChromeDriver(options);

        driver.get("https://www.sklep.pw.edu.pl/");
        wyszukiwarka = driver.findElement(By.id("woocommerce-product-search-field-0"));
        wyszukiwarka.sendKeys("koszulka");
        wyszukiwarka.sendKeys(Keys.ENTER);
        driver.findElement(By.xpath("//a[normalize-space()='Wybierz opcje']")).click();
        driver.findElement(By.id("pa_kolor")).click();
        driver.findElement(By.xpath("//select[@id='pa_kolor']/option[2]")).click();
        driver.findElement(By.id("pa_rozmiar")).click();
        driver.findElement(By.xpath("//select[@id='pa_rozmiar']/option[2]")).click();
        driver.findElement(By.xpath("//button[text()='Dodaj do koszyka']")).click();
        driver.findElement(By.xpath("//div[@class='woocommerce']//a[@class='button wc-forward']")).click();
        driver.findElement(By.xpath("//a[@class='remove']")).click();
        driver.close();
        driver.quit();
    }
}
